src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -22,6 +22,17 @@ open Fmt
 open Expression_gen
 open Statement_gen
 
+let stanc_args_to_print =
+  let sans_model_and_hpp_paths x =
+    not (String.is_suffix ~suffix:".stan" x || String.is_prefix ~prefix:"--o" x)
+  in
+  (* Ignore the "--o" arg, the stan file and the binary name (bin/stanc). *)
+  let stanc_flags_sans_model_and_hpp_paths =
+    List.filter ~f:sans_model_and_hpp_paths
+      (List.tl_exn (Array.to_list Sys.argv))
+  in
+  String.concat ~sep:" " stanc_flags_sans_model_and_hpp_paths
```

Comment:
Another way to write this that you might already know about (but just in case)

```ocaml
Array.to_list Sys.argv
|> List.tl_exn
|> List.filter ~f:sans_model_and_hpp_paths
|> String.concat ~sep:" "
```

Reply:
(also TBC I think reasonable people can have different opinions on whether the above is more or less clear! Just pointing out an option)

Reply:
Honestly, I am still not too friendly with the pipe syntax so I usually don't use it until I am at the stage of "ok now it works let's try to make it nicer". I do prefer this.